starboard/xb1/tools/xb1_launcher.py

```diff
@@ -470,6 +474,10 @@ def Run(self):
       self._LogLn('Skipping running step.')
       return 0
 
+    status, _ = self.Run2()
+    return status == ReturnCodeStatus.CRASH
+
+  def Run2(self):
     try:
       self.Kill()  # Kill existing running app.
```

Comment:
Perhaps we should try to extract the function of Kill() also into AbstractLauncher itself ? So that TestRunner will call Kill before it calls Run2() ?

Killing the previous running process is a common pattern with other launchers - e.g. Raspi-2 does the same through `pkill` calls. We should also do `adb shell am force-stop` on Android.

Reply:
I moved methods `Kill` and `SendStop` to the extended interface. Although, some implementations use special internal functions to kill the previous running process.